src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs

```diff
@@ -462,7 +462,7 @@ private XamlMemberDefinition VisitMember(XamlXmlReader reader, XamlObjectDefinit
 		private bool IsLiteralInlineText(object value, XamlMemberDefinition member, XamlObjectDefinition xamlObject)
 		{
 			return value is string
-				&& xamlObject.Type.Name == "TextBlock"
+				&& (xamlObject.Type.Name == "TextBlock" || xamlObject.Type.Name == "Span")
```

Comment:
I'm feeling this is more of a workaround. This may fail, for example, if there is a user-defined control that inherits `Span`:

```xaml
<Page
    x:Class="UWPApp.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:UWPApp"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <Grid>
        <TextBlock><local:MySpan>Hello</local:MySpan></TextBlock>
    </Grid>
</Page>
```

Anyone knows of a more elegant way?

Reply:
Yes it is a special case for sure, but that's fine for now as most use cases are going to be with non-inherited spans.